Assets/Scripts/Core/GameManager.cs

```diff
@@ -0,0 +1,180 @@
+﻿using Assets.Scripts.Model;
+using UnityEngine;
+using UnityStandardAssets.Characters.FirstPerson;
+
+public class GameManager : MonoBehaviour {
+
+    public Configurator configurator;
+
+    public Player playerPrefab;
+
+    public static GameManager instance = null;
+
+    private GameState state;
+
+    private int levelNumber;
+
+    private Player player;
+
+    // Awake is always called before any Start functions
+    void Awake()
+    {
+        if (instance == null)
+        {
+            instance = this;
+        }
+        else if (instance != this)
+        {
+            Destroy(gameObject);
+        }
+        DontDestroyOnLoad(gameObject);  // Sets this to not be destroyed when reloading scene
+
+        Init();
+
+        SwitchGameState(configurator.initialApplicationState);
+    }
+
+    private void Init()
+    {
+        levelNumber = configurator.initialLevelNumber;
+        player = Instantiate(playerPrefab) as Player;
+        player.name = "Player";
+        player.remainingLives = configurator.initialPlayerLives;
+        player.bombs = configurator.initialPlayerBombs;
+        player.bombRange = configurator.initialPlayerBombRange;
+        player.speed = configurator.initialPlayerSpeed;
+        player.remoteDetonationBonus = configurator.initialPlayerRemoteDetonationBonus;
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+
+    }
+
+    /****************************************/
+    /*          Public API Methods          */
+    /****************************************/
+
+    public void SwitchGameState(GameState state)
+    {
+        this.state = state;
+        switch (state)
+        {
+            case GameState.SPLASH:
+                SceneFader.LoadScene("Splash", 0.5f, 1);
+                // Hack - fadeOut time must be greater than zero. Otherwise animation framerate decreases significantly.
+                break;
+            case GameState.MAIN_MENU:
+                SceneFader.LoadScene("Menu", 1, 0);
+                break;
+            case GameState.SCORE_BOARD:
+                SceneFader.LoadScene("ScoreBoard", 1, 0);
+                break;
+            case GameState.GAMEPLAY:
+                SceneFader.LoadScene("Gameplay", 0, 1);
+                break;
+            default:
+                break;
+        }
+    }
+
+    public Player GetPlayer()
+    {
+        return player;
+    }
+
+    public void SetCameraRotation(Vector3 rotation)
+    {
+        FirstPersonController fpsController = player.GetComponent<FirstPersonController>();
+        fpsController.SetCameraRotation(rotation);
+    }
+
+    public void AdvanceToNextLevel()
+    {
+        levelNumber++;
+        if (levelNumber <= Configurator.MAXIMUM_LEVEL_NUMBER)
+        {
+            SwitchGameState(GameState.GAMEPLAY);
+        }
+        else
+        {
+            // WORLD EXPLODES HERE!
```

Comment:
Dlaczego w ogóle potrzebujemy ograniczać ilość leveli? Nie może być w nieskończoność? (Zakładając, że nieskończoność to jakiś Integer.MAX_VALUE)?